Katello/katello
# Fixes #34856 - Add TracerEnabler progress indicator and poll Rex jobs

- [x] fix button enablement bug
- [x] add sticky toast
- [x] un-stick toast after job succeed or fail
- [x] remove cursor that happens when mousing over disabled checkboxes

#### What are the changes introduced in this pull request?
![traces_enable](https://user-images.githubusercontent.com/22042343/166828669-22bff50e-b0ab-4480-9ef1-01ac2ab43217.png)

1. Add a loading screen for enabling Traces:

Succeed:
https://www.awesomescreenshot.com/video/8739856?key=9654f43a18a51c1da28a013fa47d4333

Fail:
https://www.awesomescreenshot.com/video/8739884?key=3e443826b2dab30f6b460dfa1f647be9

2. On all host details tabs, poll Rex jobs. While in progress, display a spinner. When they succeed, display a success toast and reload results. When they fail, display an error toast.

Package remove with results reload:
https://www.awesomescreenshot.com/video/8774035?key=6b9e8e0b8d1cf1edca62e39c91fbfb40

All this is achieved via a new `useRexJobPolling` hook.

#### Considerations taken when implementing this change?

After adding the loading indicator to Traces, I couldn't resist spreading the love!

Sadly, I was unable to use the `isLoading` prop of `<Button>` because of upcoming changes in https://github.com/Katello/katello/pull/10046. So I just added a standard spinner instead, which will allow consistency across all tabs.

#### What are the testing steps for this pull request?

* Get a working REX setup
* Enable the Satellite Tools repo so you can install Tracer (you'll probably have to override to enabled in Repo sets as well)
* New host detail page > Traces > Enable Tracer
* Make sure everything works ok and you get toasts as in the videos
* Try it again with a different host which is not running and view the failure mode

Also test the following pages and confirm that

1. you get toasts for starting, success, and failure of the job
2. results are reloaded ONLY on success (not on starting the job and not on failure)
3. Polling stops if you navigate away from the tab
4. When results are reloaded, the current search and pagination options are remembered
5. checkboxes are disabled while job is in progress, with small tooltip on mouse hover explaining the reason
6. immediate remote execution actions are disabled while job is in progress (this is to prevent users from trying the same job twice in a row, etc.)
7. customized remote execution actions remain unchanged

* Errata application
* Package remove
* Package upgrade
* Package install
* Traces restart
* Module stream actions
